packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -813,15 +801,15 @@ class PubspecYaml {
               throw StateError('Invalid pubspec.yaml: a "git" dependency section terminated early.');
             }
             line = lines[index];
-            lastDependency._lockLine += '\n$line';
+            lastDependency._lockLine = '${lastDependency._lockLine}\n$line';
           } while (line.startsWith('   '));
         }
         // We're done with this special dependency, so reset back to null so
         // we'll go in the top section next time instead.
         lastDependency = null;
       }
     }
-    return PubspecYaml._(file, packageName, packageVersion, result, checksum ?? PubspecChecksum(null, ''));
+    return PubspecYaml._(file, packageName!, packageVersion, result, checksum ?? PubspecChecksum(null, ''));
```

Comment:
Are you sure `packageName!` is safe here?

Reply:
1. it works
2. it seems to me like a bug we should fix if we don't know the packageName by this point